programs/margin-swap/src/instructions/margin_spl_swap.rs

```diff
@@ -194,10 +198,11 @@ pub struct SwapInfo<'info> {
 
 pub fn margin_spl_swap_handler(
     ctx: Context<MarginSplSwap>,
+    change_kind: ChangeKind,
     amount_in: u64,
     minimum_amount_out: u64,
 ) -> Result<()> {
-    ctx.accounts.withdraw(amount_in)?;
+    ctx.accounts.withdraw(change_kind, amount_in)?;
     ctx.accounts.swap(amount_in, minimum_amount_out)?;
```

Comment:
The `amount_in` parameter changes meaning here. When SetTo is used, it's no longer the `amount_in` quantity for a swap because for example SetTo 0 means "swap my full balance." So this line should change to two lines.

The most straightforward approach would be to copy the strategy used to determine the deposit amount at the end of this instruction. Check the balance of the transit source account and simply swap the full balance:
```suggestion
    let amount_in = token::accessor::amount(&ctx.accounts.transit_source_account)?;
    ctx.accounts.swap(amount_in, minimum_amount_out)?;
```

My only concern with this approach is that it assumes the transit account is empty to begin with, or that if there is a pre-existing balance, the user is happy to drain it for a swap. The same assumption is already built into `transit_destination_account`, so it seems OK to apply the same assumption to the source for now. Alternatively, we could check the balance before the withdrawal, and only swap the difference.